crates/starknet/src/transactions/mod.rs

```diff
@@ -0,0 +1,160 @@
+pub(crate) mod declare_transaction;
+
+use std::collections::HashMap;
+
+use starknet_api::block::BlockNumber;
+use starknet_in_rust::{execution::TransactionExecutionInfo, transaction::error::TransactionError};
+use starknet_rs_core::types::TransactionStatus;
+use starknet_types::felt::{BlockHash, TransactionHash};
+
+use crate::traits::HashIdentifiedMut;
+
+use self::declare_transaction::DeclareTransactionV1;
+
+#[derive(Default)]
+pub struct StarknetTransactions(HashMap<TransactionHash, StarknetTransaction>);
+
+impl StarknetTransactions {
+    pub fn insert(&mut self, transaction_hash: &TransactionHash, transaction: StarknetTransaction) {
+        self.0.insert(*transaction_hash, transaction);
+    }
+}
+
+impl HashIdentifiedMut for StarknetTransactions {
+    type Hash = TransactionHash;
+    type Element = StarknetTransaction;
+    fn get_by_hash_mut(&mut self, hash: &Self::Hash) -> Option<&mut StarknetTransaction> {
+        let result = self.0.get_mut(hash);
```

Comment:
Is this a remnant from debugging? If so, could be one line.

Reply:
Changed in [9aa065](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/20/commits/9aa065cb3fb1abaa9d1458a9084ce1ca0bac6ca1)